src/server/qgsconfigcache.cpp

```diff
@@ -93,6 +93,7 @@ const QgsProject *QgsConfigCache::project( const QString &path, QgsServerSetting
       QgsMessageLog::logMessage(
         QStringLiteral( "Error when loading project file '%1': %2 " ).arg( path, prj->error() ),
         QStringLiteral( "Server" ), Qgis::Critical );
+      throw QgsServerException( QStringLiteral( "Project file error when loading project file '%1': %2 " ).arg( path, prj->error() ) );
```

Comment:
No: this is disclosure of path informations.

Reply:
Ok, thought about this too. And tried to find out if I could only show the project filename... 
As said: trying to setup mapserver in osgeo4w  on windows servers with network paths' it was often hard to find what was the problem: permissions, the projectfile or something IN the projectfile..
So would only show the project file name be ok for you?

Reply:
Well, I would advice against it, if for some reason a rewrite rule was in place in order to hide the project file name we shouldn't disclose it. 

But I don't really understand why if (as you say) you are trying to ease the setup process you  don't just point the user to the server logs, that contain the full path and some more information about the various issues.

Reply:
Because (well in my experience) it is not so easy to set up the right logging on Windows when you are  not mixed Linux/Windows admin, trying to set Apache env params on a Windows machine etc etc :-)
But as said: one of my clients is bitten by this pretty often because of permission errors on the network drives where either the project files OR the data files OR they changed the db-parameters  etc etc

Reply:
Sorry, but there is no way we can disclose this kind of information to the world within the server HTLM response.